lib/Doctrine/DBAL/Driver/OCI8/OCI8Statement.php

```diff
@@ -134,7 +141,14 @@ static public function convertPositionalToNamedPlaceholders($statement)
      */
     public function bindValue($param, $value, $type = null)
     {
-        return $this->bindParam($param, $value, $type, null);
+        if ($value instanceof OCI8Descriptor) {
```

Comment:
First of all I'm glad that we could potentially still use `SQLT_CHR` :)
Second I think that the `ORA-24816` is something different that is not directly a bug of Doctrine. It looks like it is not possible to bind a non LONG column after having bound a LOB column with discriptor (sort of). I am still not completely sure what is happening here and why but I would say that is something different and should be dealt with in another PR.
Let's first focus on making DBAL OCI8 compatible with PHP7. So the only problem that is left (which I see) is the refcount thingy which we have to fix. I do not see where we have to change anything about binding types, am I right? If that is the case, could you revert everything that is not specific to keeping references to the bound values and see if our test suite succeeds? Thanks.